logviewer/src/main/java/org/togetherjava/logwatcher/views/MainLayout.java

```diff
@@ -0,0 +1,180 @@
+        DrawerToggle toggle = new DrawerToggle();
+        toggle.addClassName("text-secondary");
+        toggle.addThemeVariants(ButtonVariant.LUMO_CONTRAST);
+        toggle.getElement().setAttribute("aria-label", "Menu toggle");
+
+        viewTitle = new H1();
+        viewTitle.addClassNames("m-0", "text-l");
+
+        Header header = new Header(toggle, viewTitle);
+        header.addClassNames("bg-base", "border-b", "border-contrast-10", "box-border", "flex",
+                "h-xl", "items-center", "w-full");
+        return header;
+    }
+
+    private Component createDrawerContent() {
+        H2 appName = new H2("Logviewer");
+        appName.addClassNames("flex", "items-center", "h-xl", "m-0", "px-m", "text-m");
+
+        com.vaadin.flow.component.html.Section section = new com.vaadin.flow.component.html.Section(
+                appName, createNavigation(), createFooter());
+        section.addClassNames("flex", "flex-col", "items-stretch", "max-h-full", "min-h-full");
+        return section;
+    }
+
+    private Nav createNavigation() {
+        Nav nav = new Nav();
+        nav.addClassNames("border-b", "border-contrast-10", "flex-grow", "overflow-auto");
+        nav.getElement().setAttribute("aria-labelledby", "views");
+
+        H3 views = new H3("Views");
+        views.addClassNames("flex", "h-m", "items-center", "mx-m", "my-0", "text-s",
+                "text-tertiary");
+        views.setId("views");
+
+        for (RouterLink link : createLinks()) {
+            nav.add(link);
+        }
+        return nav;
+    }
+
+    private List<RouterLink> createLinks() {
+        return Stream
+            .of(new MenuItemInfo("Logs", "la la-globe", LogsView.class),
+                    new MenuItemInfo("Streamed", "la la-globe", StreamedView.class),
+                    new MenuItemInfo("User Management", "la la-file", UserManagement.class))
+            .filter(this::checkAccess)
+            .map(MainLayout::createLink)
+            .toList();
+    }
+
+    private boolean checkAccess(MenuItemInfo menuItemInfo) {
+        final Class<? extends Component> view = menuItemInfo.view;
+        final AllowedRoles annotation = view.getAnnotation(AllowedRoles.class);
+
+        if (annotation == null) {
+            LogManager.getLogger(MainLayout.class)
+                .warn("Class {} not properly secured with Annotation", view);
+            return false;
+        }
+
+        final Set<Role> roles = Set.of(annotation.roles());
+
+        return this.authenticatedUser.get()
+            .map(user -> !Sets.intersection(user.getRoles(), roles).isEmpty())
+            .orElse(false);
+    }
+
+    private Footer createFooter() {
+        Footer layout = new Footer();
+        layout.addClassNames("flex", "items-center", "my-s", "px-m", "py-xs");
+
+        Optional<User> maybeUser = authenticatedUser.get();
+        if (maybeUser.isPresent()) {
+            User user = maybeUser.get();
```

Comment:
Do not use `Optional#get` it will be deprecated soon. Instead:
```suggestion
            User user = maybeUser.orElseThrow();
```

Reply:
Haven't seen anything about depreciating Optional#get yet, but changed it